File: darts-ruby/spec/darts/models/throw/double_spec.rb
require 'darts'

describe Throw::Double do
  it 'should double a valid bed value' do
    expect(Throw::Double.new(10).bed_score).to eq(20)
  end

  it 'should raise exception when created with bed value just below minimum in range' do
    expect { Throw::Double.new(0) }.to raise_error(ArgumentError, 'Throw must be in range 1..20')
  end

  it 'should raise exception when created with bed value just above maximum in range' do
    expect { Throw::Double.new(21) }.to raise_error(ArgumentError, 'Throw must be in range 1..20')
  end
end

File: javascript/spec/darts/bed/double_spec.js
const Double = require('../../../app/darts/bed/double');

describe('Double', () => {
    describe('.bedScore', () => {
        it('should double a valid bed score', () => {
            expect(new Double(10).bedScore()).toEqual(20);
        });

        it('should raise exception when created with bed score just below minimum in range', () => {
            expect(() => { new Double(0) }).toThrowError(RangeError, 'Bed score must be in range 1..20');
        });
    });
});
